Add borderRadius to InkWell in TabBar using provided parameter

nayeemtby opened the pull request:
This PR is adding splashBorderRadius to TabBar that provides borderRadius to InkWell.

Addresses Feature Request #94290

Before: https://youtu.be/S2qBpaE2wOo
After: https://youtu.be/xvK_uZXVmDc

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

nayeemtby committed:
Initial Commit

nayeemtby committed:
Initial Commit

nayeemtby committed:
Initial Commit

nayeemtby committed:
Initial Commit

nayeemtby committed:
Fixed issues with test

nayeemtby committed:
Fixed issues with test

nayeemtby committed:
Fixed issues with test

nayeemtby committed:
Probably fixed Doc issues.

nayeemtby committed:
Probably fixed Doc issues.

nayeemtby committed:
Probably fixed Doc issues.

nayeemtby committed:
Probably fixed Doc issues.

nayeemtby commented:
Requesting any recommendation regarding the parameter name.

nayeemtby commented:
Any recommendation on the parameter name?

innos123 commented:
> This PR is adding splashBorderRadius to TabBar that provides borderRadius to InkWell.
> 
> Addresses Feature Request #94290
> 
> Before: https://youtu.be/S2qBpaE2wOo After: https://youtu.be/xvK_uZXVmDc
> 
> ## Pre-launch Checklist
> * [x]  I read the [Contributor Guide](https://github.com/flutter/flutter/wiki/Tree-hygiene#overview) and followed the process outlined there for submitting PRs.
> * [x]  I read the [Tree Hygiene](https://github.com/flutter/flutter/wiki/Tree-hygiene) wiki page, which explains my responsibilities.
> * [x]  I read and followed the [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo), including [Features we expect every widget to implement](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement).
> * [x]  I signed the [CLA](https://cla.developers.google.com/).
> * [x]  I listed at least one issue that this PR fixes in the description above.
> * [x]  I updated/added relevant documentation (doc comments with `///`).
> * [x]  I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests).
> * [x]  All existing and new tests are passing.
> 
> If you need help, consider asking for advice on the #hackers-new channel on [Discord](https://github.com/flutter/flutter/wiki/Chat).

werainkhatri commented on packages/flutter/lib/src/material/tabs.dart:
```diff
@@ -832,6 +833,12 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
   /// Defaults to matching platform conventions.
   final ScrollPhysics? physics;
 
+  /// Defines the [InkWell]'s `borderRadius` property.
```
The doc should not reference something the developer may not know about, like `TabBar` using `InkWell`. Try documenting it like below.

```dart
/// The clipping radius of splashes that extend outside the bounds of the tab.
```

Try reading docs of similar API from other parts of the framework to get a better idea.

werainkhatri commented on packages/flutter/lib/src/material/tabs.dart:
```diff
@@ -832,6 +833,12 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
 
+  /// Defines the [InkWell]'s `borderRadius` property.
+  ///
+  /// For example, to match the `borderRadius` of [BoxDecoration] provided as
+  /// `indicator` to this [TabBar].
```
Instead of here, add `splashBorderRadius` in the `See also` section of the doc of `indicator`, explaining this point there.

werainkhatri replied:
Remove this from here as you have added it in the see also section.

werainkhatri commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -4183,6 +4183,40 @@ void main() {
+    await gesture.moveTo(tester.getCenter(find.byType(Tab)));
+    await tester.pumpAndSettle();
+    final RenderObject object = tester.allRenderObjects.firstWhere((RenderObject element) => element.runtimeType.toString() == '_RenderInkFeatures');
+    expect( object, paints..rrect(
+                                color: _hoverColor,
+                                rrect: RRect.fromRectAndRadius(
+                                    tester.getRect(find.byType(InkWell)),
+                                    const Radius.circular(_radius))));
```
The formatting needs correction. Refer to the Style Guide or other tests to know more.

werainkhatri commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -4183,6 +4183,40 @@ void main() {
+                  Tab(
+                    child: Text(''),
+                  )
+                ]))));
```
The formatting needs correction.

nayeemtby committed:
Changed docs, added see also in indicator doc

nayeemtby committed:
Changed docs, added see also in indicator doc

nayeemtby committed:
Changed docs, added see also in indicator doc

nayeemtby committed:
added null indication

nayeemtby committed:
added null indication

nayeemtby committed:
added null indication

nayeemtby committed:
added null indication

nayeemtby committed:
Changed formatting

nayeemtby committed:
Changed formatting

nayeemtby committed:
Changed formatting

nayeemtby committed:
Grammer check

nayeemtby committed:
Grammer check

nayeemtby committed:
Grammer check

nayeemtby committed:
Grammer check

nayeemtby committed:
Removed trailing space

nayeemtby committed:
Removed trailing space

nayeemtby committed:
Removed trailing space

nayeemtby commented:
Is the plugins error related to my PR?

werainkhatri commented:
Could you try to rebase/merge this with the master branch? Looks like there are conflicts too.

nayeemtby committed:
Resolve Conflict

nayeemtby committed:
Fixed a oops! mistake

nayeemtby commented:
> Could you try to rebase/merge this with the master branch? Looks like there are conflicts too.

Done

werainkhatri commented on packages/flutter/lib/src/material/tabs.dart:
```diff
@@ -719,6 +720,10 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
+  ///
+  /// See also:
+  ///
+  ///   * [splashBorderRadius] to define the clipping radius of the splash if borderRadius of BoxDecoration is defined.
```
```suggestion
  ///  * [splashBorderRadius], which defines the clipping radius of the splash
  ///    and is generally used with [BoxDecoration.borderRadius].
```

werainkhatri commented on packages/flutter/lib/src/material/tabs.dart:
```diff
@@ -832,6 +837,14 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
+  /// For example, to match the `borderRadius` of [BoxDecoration] provided as
+  /// `indicator` to this [TabBar].
+  ///
+  /// If this property is null, then [BorderRadius.zero] is used.
```
```suggestion
  /// If this is null, it is interpreted as [BorderRadius.zero].
```

werainkhatri replied:
```suggestion
  /// When null, it is interpreted as [BorderRadius.zero].
```

werainkhatri commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -4256,6 +4256,50 @@ void main() {
+            tabs: const <Widget>[
+              Tab(
+                child: Text(''),
+              )
+            ]
+          )
+        )
+      )
```
```suggestion
              ),
            ],
          ),
        ),
      ),
```

werainkhatri commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -4256,6 +4256,50 @@ void main() {
+        rrect: RRect.fromRectAndRadius(
+          tester.getRect(find.byType(InkWell)),
+          const Radius.circular(_radius)
+        )
+      )
```
```suggestion
        ),
      ),
```

werainkhatri commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -4256,6 +4256,50 @@ void main() {
     expect(pageController.page, 2);
   });
+
+  testWidgets("TabBar accepts 'splashBorderRadius' that gives 'borderRadius' to InkWell", (WidgetTester tester) async {
```
```suggestion
  testWidgets("splashBorderRadius is passed to InkWell.borderRadius", (WidgetTester tester) async {
```

nayeemtby committed:
Merge branch 'flutter:master' into Add-TabBar-indicator-splash-radius

nayeemtby committed:
Applied Suggestions

nayeemtby committed:
Applied Suggestions

nayeemtby committed:
Applied Suggestions

nayeemtby committed:
Applied Suggestions

nayeemtby committed:
Replaced double quotes

nayeemtby committed:
Replaced double quotes

nayeemtby committed:
Replaced double quotes

nayeemtby committed:
Merge branch 'flutter:master' into Add-TabBar-indicator-splash-radius

chunhtai reviewed (approved):
LGTM

chunhtai commented:
can you rebase off the latest master and see if it fixes the build?

nayeemtby committed:
Merge branch 'flutter:master' into Add-TabBar-indicator-splash-radius

nayeemtby commented:
> can you rebase off the latest master and see if it fixes the build?

Done, but I think the problem is internal. 

> /root/.pub-cache/hosted/pub.dartlang.org/platform-2.2.1/lib/src/interface/local_platform.dart:45:41: Error: Member not found: 'packageRoot'.
|   String get packageRoot => io.Platform.packageRoot;

This error is all over the place.

nayeemtby committed:
Merge branch 'master' into Add-TabBar-indicator-splash-radius

nayeemtby committed:
Added missing tokens after merge

nayeemtby committed:
Fixed another mistake

nayeemtby committed:
Added missing tokens

nayeemtby committed:
Merge branch 'flutter:master' into Add-TabBar-indicator-splash-radius

nayeemtby committed:
Merge branch 'Add-TabBar-indicator-splash-radius' of https://github.com/nayeemtby/flutter into Add-TabBar-indicator-splash-radius

engine-flutter-autoroll committed:
Roll Plugins from de267d35fbb4 to 2d0cec9fb4a3 (1 revision) (#97006)

engine-flutter-autoroll committed:
Roll Engine from 665845d9f1de to 0ca70b6b9677 (1 revision) (#97009)

godofredoc committed:
Re-enable plugins_analyze. (#96977)

The test was disabled for release branches because there were some
failures that got attributed to checking plugins from master.

Checking the logic of the test more closely it is already using a pinned
version of the plugins repository and it should be consistent on
pre/post submit validations.

Bug: https://github.com/flutter/flutter/issues/91935

WasserEsser committed:
Call image stream listeners asynchronously if added asynchronously (#95525)

* Call image stream listeners with new flag

* Make _synchronousCall private

* Rename _synchronousCall parameter to be more descriptive and add more documentation

* Rename _hasInitialListeners to _addingInitialListeners and provide a better explanation

* Formatting fix

* Add tests

* Remove dependency on decoration_test.dart

* Simplify tests

* Remove empty line in comment

* Documentation

* Trigger tests

* Trigger tests

* Flip boolean value

chunhtai commented:
Looks like there is a bunch formatting error, can you fix them as well?

albertodev01 committed:
chore: Moved didUpdateWidget calls first (#96944)

engine-flutter-autoroll committed:
Roll Engine from 0ca70b6b9677 to 136cafa6b43a (1 revision) (#97014)

fluttergithubbot committed:
Marks Linux_android flutter_gallery__back_button_memory to be unflaky (#96065)

engine-flutter-autoroll committed:
Roll Engine from 136cafa6b43a to d26defded430 (1 revision) (#97025)

fluttergithubbot committed:
Marks Linux deferred components to be flaky (#96404)

TahaTesser committed:
Explain how MaterialApp renders text style in absence of Material Widget (#96530)

keyonghan committed:
Marks Mac_ios native_ui_tests_ios to be flaky (#97026)

engine-flutter-autoroll committed:
Roll Engine from d26defded430 to 2a13912986a7 (1 revision) (#97027)

goderbauer committed:
Enable no_leading_underscores_for_local_identifiers (#96422)

engine-flutter-autoroll committed:
Roll Engine from 2a13912986a7 to 3a116fea6a21 (2 revisions) (#97030)

engine-flutter-autoroll committed:
Roll Plugins from 2d0cec9fb4a3 to 3fb9e2c13b40 (2 revisions) (#97033)

TahaTesser committed:
Add ability to control if bottom child focus can be excluded in `AnimatedCrossFade` (#96593)

engine-flutter-autoroll committed:
Roll Engine from 3a116fea6a21 to 0ff9e89991a7 (3 revisions) (#97035)

engine-flutter-autoroll committed:
Roll Engine from 0ff9e89991a7 to ded447844f3e (4 revisions) (#97040)

engine-flutter-autoroll committed:
Roll Engine from ded447844f3e to 18cbe9724d6b (1 revision) (#97044)

engine-flutter-autoroll committed:
Roll Engine from 18cbe9724d6b to 5061c0fb74cf (2 revisions) (#97045)

pedromassango committed:
Add closeDrawer and closeEndDrawer in ScaffoldState (#96960)

engine-flutter-autoroll committed:
Roll Engine from 5061c0fb74cf to 5b2068872afe (1 revision) (#97049)

engine-flutter-autoroll committed:
Roll Engine from 5b2068872afe to 901b2787c8d6 (5 revisions) (#97069)

nt4f04uNd committed:
PageView scroll physics to match Android (#95423)

engine-flutter-autoroll committed:
Roll Engine from 901b2787c8d6 to b8ad3d4787c7 (2 revisions) (#97075)

engine-flutter-autoroll committed:
Roll Engine from b8ad3d4787c7 to 0fa807085f04 (1 revision) (#97081)

engine-flutter-autoroll committed:
Roll Engine from 0fa807085f04 to 51a54c2b729a (2 revisions) (#97085)

engine-flutter-autoroll committed:
Roll Engine from 51a54c2b729a to 8162e8d86e47 (1 revision) (#97093)

engine-flutter-autoroll committed:
Roll Engine from 8162e8d86e47 to 91637bf4479a (1 revision) (#97096)

engine-flutter-autoroll committed:
Roll Engine from 91637bf4479a to 4178ff3dd587 (1 revision) (#97098)

engine-flutter-autoroll committed:
Roll Engine from 4178ff3dd587 to 4f58a01268a5 (1 revision) (#97099)

engine-flutter-autoroll committed:
Roll Engine from 4f58a01268a5 to 8068203f5f85 (1 revision) (#97114)

engine-flutter-autoroll committed:
Roll Engine from 8068203f5f85 to 8204850dbee1 (3 revisions) (#97122)

engine-flutter-autoroll committed:
Roll Engine from 8204850dbee1 to e920762d7011 (1 revision) (#97128)

engine-flutter-autoroll committed:
Roll Engine from e920762d7011 to ff799f4b1862 (1 revision) (#97129)

TahaTesser committed:
ListTile: add themeable mouse cursor  (#96740)

werainkhatri committed:
Adds `CheckboxListTile.checkboxShape` (#95714)

zanderso committed:
Adjust lipo thinning input/output for macOS (#97111)

zanderso committed:
Use frontend_server from the Dart SDK (#97078)

chunhtai committed:
Allow current leader of layerlink to detach before the previous leade… (#96810)

stuartmorgan-g committed:
Add a note about depot_tools to the `led` step (#96830)

mateusfccp committed:
Export shadows to `Icon` API (#83638)

Piinks committed:
Deprecate Scrollbar isAlwaysShown -> thumbVisibility (#96957)

engine-flutter-autoroll committed:
a7a32daa9 Fix Memory Leak in MakeSkSurfaceFromBackingStore (flutter/engine#30735) (#97147)

engine-flutter-autoroll committed:
Roll Engine from a7a32daa90ec to 83cfdcc8f101 (2 revisions) (#97164)

* 016f45840 Roll Skia from 67f92df4b059 to b894b0594eee (1 revision) (flutter/engine#31019)

* 83cfdcc8f [web] Remove usage of flutter/goldens (flutter/engine#30116)

LongCatIsLooong committed:
Show keyboard after text input connection restarts (#96541)

Piinks committed:
Revert "PageView scroll physics to match Android (#95423)" (#97150)

This reverts commit daef08253b94f5fd1e188e16fc0d9524d4f426d2.

Commit:
[framework] remove extra casts from framework (#97155)

Remove cast from hit_test matrix code, linkedhashmap cast, and insertion sort cast

Jasguerrero committed:
[flutter_tools] support files in flutter analyze #96231 (#97021)

engine-flutter-autoroll committed:
Roll Plugins from 3fb9e2c13b40 to d6619d8e8b47 (1 revision) (#97153)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_one_rect_perf__e2e_summary to be unflaky (#97137)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_col_of_rows_perf__e2e_summary to be unflaky (#97138)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_opacity_of_grid_perf__e2e_summary to be unflaky (#97139)

fluttergithubbot committed:
Marks Mac module_test_ios to be unflaky (#97145)

Dan-Crane committed:
add direction to CupertinoPickerDefaultSelectionOverlay (#92959)

chunhtai committed:
relax routerReportsNewRouteInformation function signature (#97154)

engine-flutter-autoroll committed:
Roll Plugins from d6619d8e8b47 to c0b403222a4f (2 revisions) (#97176)

stuartmorgan-g committed:
Don't use example folder as a project type signal (#97157)

The code to generate platform files used to use the presence of an
'example' folder to determine whether or not a project was a plugin (in
which case that generation is bypassed). Later, robust detection was
added based on the pubspec.yaml `plugin` key, but the old check was left
in place. This creates false positives where people who add an example
folder to their app project start getting mysterious failures.

Since the pubspec check is definitive, there's no reason to continue to
use the presence of `example` as an indicator at all.

Fixes https://github.com/flutter/flutter/issues/87007

jmagman committed:
Move gallery native UI test launching out of test and into setup (#97177)

nayeemtby committed:
Fix rebase issues

nayeemtby committed:
Merge branch 'Add-TabBar-indicator-splash-radius' of https://github.com/nayeemtby/flutter into Add-TabBar-indicator-splash-radius

nayeemtby committed:
changed variable identifier
